### PR TITLE
Kafka tiered storage concept content

### PR DESCRIPTION
**Pull request by harshini-rangaswamy**
**NOTE: PLEASE DO NOT MERGE YET**

# What changed, and why it matters

Aiven for Apache Kafka tiered storage concept content: 
 - Overview
 - How it works 
 - Guarantee
 - Backups
 - Limitations

### CODE REVIEW

**Review by jeqo — APPROVED** [review]
LGTM, thanks @harshini-rangaswamy!

File: docs/products/kafka/concepts/tiered-storage-how-it-works.rst [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+--------------------------------------
+Data is transferred to remote storage asynchronously and does not interfere with the producer activity. While the broker aims to move data as swiftly as possible, certain conditions, such as high-throughput or connectivity issues, may cause more data to be stored in the local storage than the specified local retention policy.
+
+Any data exceeding the local retention threshold will not be purged by the log cleaner until it is successfully uploaded to remote storage.
+The replication factor is not considered during the upload process, and only one copy of each segment is uploaded to the remote storage. Most remote storage options have their own measures, including data replication, to ensure data durability.
+
+
+Data retrieval
+-----------------
+When consumers fetch records stored in remote storage, the broker downloads and caches these records locally. This allows for quicker access in subsequent retrieval operations.
```
Comment (jeqo) [inline docs/products/kafka/concepts/tiered-storage-how-it-works.rst:38]: Include comment about local disk cache:
```suggestion
When consumers fetch records stored in remote storage, the broker downloads and caches these records locally. This allows for quicker access in subsequent retrieval operations.
Aiven will use a small amount of disk (between 2GB and 16GB, based on the 5% of disk available on the broker) to store records fetched temporally.
```

File: docs/products/kafka/concepts/tiered-storage-how-it-works.rst [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  :alt: Diagram depicting the concept of local vs. remote data retention in a tiered storage system.
+
+Segment management
+-------------------
+Data is organized into segments, which are uploaded to remote storage individually. The active (newest) segment remains in local storage, which means that the segment size can also influence local data retention. For instance, if the local retention threshold is 1 GB, but the segment size is 2 GB, the local storage will exceed the 1 GB limit until the active segment is rolled over and uploaded to remote storage.
+
+
+Asynchronous uploads and replication
+--------------------------------------
+Data is transferred to remote storage asynchronously and does not interfere with the producer activity. While the broker aims to move data as swiftly as possible, certain conditions, such as high-throughput or connectivity issues, may cause more data to be stored in the local storage than the specified local retention policy.
```
Comment (ivanyu) [inline docs/products/kafka/concepts/tiered-storage-how-it-works.rst:30]:
```suggestion
Data is transferred to remote storage asynchronously and does not interfere with the producer activity. While the broker aims to move data as swiftly as possible, certain conditions, such as high ingestion rate or connectivity issues, may cause more data to be stored in the local storage than the specified local retention policy.
```

**Review by ivanyu — CHANGES_REQUESTED** [review]
One small comment

**Pull request merged**